tools/klee/main.cpp

```diff
@@ -421,7 +421,7 @@ void KleeHandler::processTestCase(const ExecutionState &state,
                                   const char *errorMessage,
                                   const char *errorSuffix) {
   if (errorMessage && ExitOnError) {
-    llvm::errs() << "EXITING ON ERROR:\n" << errorMessage << "\n";
+    klee_error("EXITING ON ERROR:\n%s\n", errorMessage);
```

Comment:
`klee_error` exits, you should thus move `m_interpreter->prepareForEarlyExit();` before and remove `exit(1)`